pyproject.toml

```diff
@@ -8,6 +8,7 @@ dependencies = [
     "langchain-core",
     "langchain-community",
     "langchain_openai",
+    "llama_index",
```

Comment:
llamaindex is actually not part of the core dependency. It's best if we keep it optional

Reply:
if you move it to `all` that would be helpful, or maybe even a new group for llamaindex?

Reply:
Thanks for the feedback! I've updated the PR to make llama_index optional by moving it to the [all] group.